### ISSUE TITLE
[Bug]: result is not correct when varchar equivalent query

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#14624]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

aa9296994ca4244ce70f8ea1f433e742319371ae

### Other Environment Information

```Markdown
- Hardware parameters: 64C 256G
- OS type:
- Others:
```

### Actual Behavior

**the result is not correct for varchar equivalent queries:**

mysql> select vc_code from ecbase_card ec WHERE ec.vc_code = 'ZBBKCNBZ451';
^C^C -- query aborted
+-------------+
| vc_code     |
+-------------+
| PBOCCNBJ771 |
+-------------+
9388032 rows in set (2.76 sec)

two problem:
1. the real count of dataset is only 294;
2. the only value in this query result is **'ZBBKCNBZ451'**, should never contain **'PBOCCNBJ771'** 

mo-log:
[mo-service-count-error.tar.gz](https://github.com/matrixorigin/matrixone/files/14359983/mo-service-count-error.tar.gz)

### Steps to Reproduce

```Markdown
DDL:
CREATE TABLE `ecbase_card` (
`id` BIGINT NOT NULL AUTO_INCREMENT COMMENT '自增主键',
`company_id` BIGINT NOT NULL COMMENT '所属企业Id，作为分区字段，与自增长id作为联合主键',
`code` VARCHAR(32) DEFAULT null COMMENT '车企编码 kafak导入',
`platform_id` VARCHAR(64) DEFAULT null COMMENT '平台ID',
`subs_id` VARCHAR(32) DEFAULT null COMMENT '用户id kafak导入',
`uuid` VARCHAR(64) NOT NULL COMMENT '唯一标识',
`create_time` DATETIME NOT NULL COMMENT '创建时间',
`create_by` VARCHAR(255) DEFAULT null COMMENT '创建者',
`update_time` DATETIME DEFAULT null COMMENT '更新时间',
`update_by` VARCHAR(255) DEFAULT null COMMENT '更新者',
`iccid` VARCHAR(32) NOT NULL COMMENT 'ICCID',
`imsi` VARCHAR(32) DEFAULT null COMMENT 'IMSI kafak导入',
`msisdn` VARCHAR(32) DEFAULT null COMMENT 'IMSI kafak导入',
`carrier` INT NOT NULL COMMENT '所属运营商',
`imei` VARCHAR(32) DEFAULT null COMMENT 'IMEI',
`vin` VARCHAR(32) DEFAULT null COMMENT '车辆VIN码',
`open_time` DATETIME DEFAULT null COMMENT '开户日期',
`active_date` DATETIME DEFAULT null COMMENT '激活日期',
`network_type` VARCHAR(8) DEFAULT null COMMENT '网络类型（字典项编码）',
`card_type` VARCHAR(8) DEFAULT null COMMENT '卡片物理类型（字典项编码）',
`belong_place` VARCHAR(32) DEFAULT null COMMENT '归属地',
`remark` VARCHAR(255) DEFAULT null COMMENT '备注',
`card_status` VARCHAR(8) DEFAULT null COMMENT '卡状态字典项编码，根据平台区分 kafak导入',
`status_time` DATETIME DEFAULT null COMMENT '卡号状态变更时间',
`vehicle_status` INT DEFAULT null COMMENT '车辆状态',
`vehicle_out_factory_time` DATETIME DEFAULT null COMMENT '车辆出厂时间',
`realname_status` INT DEFAULT '0' COMMENT '实名状态：0-初始化未实名，1-受理成功处理中，2-实名登记成功，3-实名注销成功',
`realname_status_t1` INT DEFAULT null COMMENT 't1实名状态：1：已实名，2：未实名',
`pay_account_id` VARCHAR(32) DEFAULT null COMMENT '付费账户id',
`pay_account_name` VARCHAR(64) DEFAULT null COMMENT '付费账户名称',
`plat_type` VARCHAR(8) DEFAULT null COMMENT '平台类型2:pb 3:ct',
`cust_id` VARCHAR(32) DEFAULT null COMMENT '客户id kafak导入',
`cust_name` VARCHAR(32) DEFAULT null COMMENT '客户名称',
`cust_type` VARCHAR(32) DEFAULT null COMMENT '客户类型 一般为C客户类型',
`be_id` VARCHAR(32) DEFAULT null COMMENT '省份编码',
`region_id` VARCHAR(32) DEFAULT null COMMENT '归属地编码',
`group_id` VARCHAR(128) DEFAULT null COMMENT '归属群组id',
`group_member_status` VARCHAR(128) DEFAULT null COMMENT '归属群组中成员状态',
`group_pool_data_usage` BIGINT DEFAULT null COMMENT '归属群组中本月池内用量',
`cust_code` VARCHAR(255) DEFAULT null COMMENT '客户集团编码',
`sync_time` DATETIME DEFAULT null COMMENT '通过kafka入库时，每次必须更新的字段，其他入口不用变动',
`boss` INT NOT NULL COMMENT '运营商BOSS系统，1-移动CT，2-移动PB，3-电信DCP，4-电信M2M，5-联通jasper，6-联通CMP，7-中国电信5GCMP平台',
`account_id` BIGINT DEFAULT null COMMENT 'BOSS系统账号id',
`vc_code` VARCHAR(32) DEFAULT null COMMENT '车企vc_code',
`source_create_time` DATETIME DEFAULT null COMMENT '源端数据创建时间',
`source_modify_time` DATETIME DEFAULT null COMMENT '源端数据修改时间',
`device_num` VARCHAR(32) DEFAULT null COMMENT '设备号',
PRIMARY KEY (`id`,`company_id`),
UNIQUE KEY `uk_ecbase_card_uuid` (`uuid`),
UNIQUE KEY `uk_ecbase_card_vc_code_iccid` (`vc_code`,`iccid`) COMMENT '20231021新增',
KEY `ecbase_card_ecbase_company_id_fk` (`company_id`),
KEY `idx_ecbase_card_create_time` (`create_time`),
KEY `idx_account_id` (`account_id`),
KEY `ecbase_card_subs_id` (`subs_id`) COMMENT '20230529新增',
KEY `idx_ecbase_card_msisdn` (`msisdn`),
KEY `ecbase_card_vc_code_carrier_card_status_idx` (`vc_code`,`carrier`,`card_status`),
KEY `ecbase_card_vc_code_vehicle_status_idx` (`vc_code`,`vehicle_status`),
KEY `ecbase_card_vc_code_realname_status_idx` (`vc_code`,`realname_status`),
KEY `ecbase_card_vin_idx` (`vin`),
KEY `ecbase_card_iccid` (`iccid`) COMMENT '2310.7新增'
) COMMENT='卡信息表';

the table data in on 129 svc, pwd: /data1/sudong/mock-data-generator/data/oneiov
```

**Comment by sukki37** [comment]
maybe dup with #14624

**Comment by XuPeng-SH** [comment]
@w-zr please help to investigate this issue.

**Comment by w-zr** [comment]
![image](https://github.com/matrixorigin/matrixone/assets/28624654/ae1d58b9-c0e6-4e46-a861-3fa75eee8a93)
result of runtime filter is correct(correct result is 294).

**Comment by w-zr** [comment]
already fixed by #14602

**Comment by aressu1985** [comment]
fixed
